Sources/XiCli/CommandLineTool.swift

```diff
@@ -0,0 +1,45 @@
+// Copyright 2018 The Xi-Editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+
+public final class CommandLineTool {
+    private let arguments: [String]
+    
+    public init(arguments: [String] = CommandLine.arguments) {
+        self.arguments = arguments
+    }
+    
+    public func run() throws {
+        guard arguments.count > 1 else {
+            throw Error.missingFileName
+        }
+        // The first argument is the execution path
+        let fileName = arguments[1]
+        
+        if(FileManager.default.fileExists(atPath: fileName)){
+            print("opening \(fileName)...")
+            NSWorkspace.shared.openFile(fileName, withApplication: "XiEditor")
+        } else {
+            throw Error.fileDoesNotExist
```

Comment:
nit: I think this `throw` would look great in a `guard` statement :)

Reply:
Like?
```suggestion
      guard FileManager.default.fileExists(atPath: fileName) else {
            throw Error.fileDoesNotExist
       }
       print("opening \(fileName)...")
       NSWorkspace.shared.openFile(fileName, withApplication: "XiEditor")
```

Reply:
exactly ^^

Reply:
Amended with that change!